### PR TITLE
WriteAdditionalData Code Cleanup

### PR DESCRIPTION
**Pull request by dadams39**
- Initial look at adding maps to the writer to support mapping of  Primitive types.
- General code clean-up to add a switch statement

### CODE REVIEW

**Comment by dadams39** [comment]
### NOTE 

The original code had this unreachable code:
https://github.com/microsoft/kiota-serialization-json-go/blob/e5b31bc06f4f1591c632b927e404ab1b69528065/json_serialization_writer.go#L920

because of ->  https://github.com/microsoft/kiota-serialization-json-go/blob/e5b31bc06f4f1591c632b927e404ab1b69528065/json_serialization_writer.go#L624
The endpoints for the clauses are different. This was one of the major reasons for the code restructure. The other was to ensure that trusted functions which were already created could be reused.

File: json_serialization_writer.go [hunk 1]
```diff
@@ -585,350 +585,232 @@ func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
...
+		err = w.WriteISODurationValue(key, &value)
+	case *absser.TimeOnly:
+		err = w.WriteTimeOnlyValue(key, value)
+	case absser.TimeOnly:
+		err = w.WriteTimeOnlyValue(key, &value)
+	case *absser.DateOnly:
+		err = w.WriteDateOnlyValue(key, value)
+	case absser.DateOnly:
+		err = w.WriteDateOnlyValue(key, &value)
+	case *JsonParseNode:
```
Comment (dadams39) [inline json_serialization_writer.go:688]: This was a generic way to add the capability needed for `dateTime` and `timeZone` functionality. A more robust handler for these types could be created and tested if the parsing of eventMessageRequest is going to remain the same.

**Comment by dadams39** [comment]
**STATUS** `WIP`
Mapping functions requested to be removed from writer source code. See Issue #28.  Wll switch the status to `R4R` when changes are submitted.

**Comment by dadams39** [comment]
**Status:** `R4R`

File: json_serialization_writer.go [hunk 2]
```diff
@@ -585,352 +585,112 @@ func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
...
 			}
-			return fmt.Errorf("unsupported AdditionalData type: %T", value)
 		}
 	}
 	return nil
 }
 
+// writeAdditional Helper function for writing additional data. Feature adds
+// the ability to support certain map types.
+func (w *JsonSerializationWriter) writeAdditional(key string, input any) error {
```
Comment (baywet) [inline json_serialization_writer.go:603]: now that we're back to a single method, would you mind moving the body back to the public method and removing this private one?

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for making the updates, one last comment

File: json_serialization_writer.go (hunk 2 above)
Comment (dadams39) [reply json_serialization_writer.go]: Changes committed.

File: json_serialization_writer.go (hunk 1 above)
Comment (dadams39) [reply json_serialization_writer.go]: Decided that this would not be implemented in this branch. See Issue #30

**Review by baywet — CHANGES_REQUESTED** [review]

**Pull request merged**